go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -152,14 +153,25 @@ func (tm *TabletManager) restoreDataLocked(ctx context.Context, logger logutil.L
 	if err != nil {
 		return err
 	}
+
+	startTime := logutil.ProtoToTime(keyspaceInfo.SnapshotTime)
+
+	// Check if we should use the latest (default) or a specified backup timestamp for the restore
+	if restoreFromBackupTs != "" {
+		startTime, err = time.Parse(mysqlctl.BackupTimestampFormat, restoreFromBackupTs)
+		if err != nil {
+			return vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, fmt.Sprintf("unable to parse the backup timestamp value provided of '%s'", restoreFromBackupTs))
+		}
+	}
```

Comment:
I skimmed the rest of the PR and it doesn't seem like anyone brought this up. Another way to do this would be to push the parsing further up to the cli, and enforce stricter types closer to this function. In this way we could:
* `commandRestoreBackup` in `go/vt/vtctl/vtctl.go` parses its `backupTimestampStr` into a `time.Time`, using `mysqlctl.BackupTimestampFormat`
* we pass `time.Time` to all the tmc functions
* when making the gRPC request, we change that protobuf definition to be a `vttime.Time`, and call `protoutil.TimeToProto` to convert the `time.Time` to a `vttime.Time`, and convert back on the tabletmanager side.

What do you think?

Reply:
I briefly considered this but wasn't sure of the benefits (avoiding the RPC call with an invalid vtctl CLI parameter is one). I don't mind making this change if you prefer it? Sounds like you do or you wouldn't have mentioned it 😄 ... so I'll work on that now. Thanks!

Reply:
OK, after trying it again I'm remembering WHY I didn't go this route last time. 🙂

What makes it feel awkward/clunky to use `time.Time` coming from the `vtctl` side is that there's also a corresponding `vttablet` flag that's used directly in the tmc: https://github.com/vitessio/vitess/blob/a7a951beeee056af478958f233c81dbe417c33fb/go/vt/vttablet/tabletmanager/restore.go#L49

As things are now, it's uniform handling (always a string) on the tmc side that we convert when using -- whether the value came from a vtctl call and RPC or from a vttablet flag. Maybe I'm missing something though?

I understand the pull to use a more precise type... but it also feels like it makes the code more complex with little practical benefit. If you feel strongly about it though I can continue.

Reply:
OK, I just switched to this method. Thank you for the nudge! I feel better about it too and It wasn't really that awkward in the end. :-)